File: packages/native_crypto_android/android/src/main/kotlin/fr/pointcheval/native_crypto_android/HashAlgorithm.kt
package fr.pointcheval.native_crypto_android

enum class HashAlgorithm(val length : Int) {
    SHA256(256),
    SHA384(384),
    SHA512(512);

    fun hmac(): String {
        return when (this) {
            HashAlgorithm.SHA256 -> "HmacSHA256"
            HashAlgorithm.SHA384 -> "HmacSHA384"
            HashAlgorithm.SHA512 -> "HmacSHA512"
        }
    }
}

File: packages/native_crypto/lib/src/core/enums/hash_algorithm.dart
// Copyright 2019-2023 Hugo Pointcheval
//
// Use of this source code is governed by an MIT-style
// license that can be found in the LICENSE file or at
// https://opensource.org/licenses/MIT.

/// The hash algorithm to use in Message Digest and HMAC.
enum HashAlgorithm {
  /// The SHA-256 hash algorithm.
  sha256,

  /// The SHA-384 hash algorithm.
  sha384,

  /// The SHA-512 hash algorithm.
  sha512,
}
